File: typescript/outputDevices.ts
export default abstract class OutputDevice{
    abstract output(): boolean;
}

export class Monitor extends OutputDevice{
    output(){
        console.log("Outputing data on the Monitor...");
        return true;
    }
}

export class Mouse extends OutputDevice{
    output(){
        console.log("Outputing data on Mouse...");
        return true;
    }
}

File: python/output_devices.py
from abc import ABC, abstractmethod
from typing import Any

class OutputDevice(ABC):
    @abstractmethod
    def output(self, data: Any) -> bool:
        pass

class Monitor(OutputDevice):
    def output(self, data: Any) -> bool:
        print("=======================================")
        print("Outputting data on the Monitor Screen...")
        print("=======================================")
        print("Data:")
        print(data)
        print("=======================================")
        return True

class Projector(OutputDevice):
    def output(self, data: Any) -> bool:
        print("=======================================")
        print("Outputting data on a Projection...")
        print("=======================================")
        print("Data:")
        print(data)
        print("=======================================")
        return True
